factorial/factorial.c

```diff
@@ -0,0 +1,45 @@
+#include <inttypes.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+// Signatures of functions defined later
+int validateInput(int, char **);
+uintmax_t factorial(uintmax_t);
+
+// Entrypoint
+int main( int argc, char *argv[] )  {
+    uintmax_t value = validateInput(argc, argv);
+    printf("Input value: %lu\n", value);
+    uintmax_t result = factorial(value);
+    printf("     Result: %lu\n", result);
+    return 0;
+}
+
+// Calculate factorial
+uintmax_t factorial(uintmax_t value) {
+    if (value < 2) {
+        return 1;
+    }
+    uintmax_t counter = value;
+    uintmax_t result = value;
+    while (counter > 1) {
+        result = result * (counter - 1);
+        counter--;
+    }
+    return result;
+}
+
+// Validate input parameters
+int validateInput(int argc, char *argv[]) {
+    if (argc != 2) {
+        puts("One argument expected.");
+        exit(1);
```

Comment:
I think this is fine for a small program like this, but in general I would be hesitant to employ the `exit` method outside of anything but your topmost functions (like main).

Reply:
You could decouple (into separate methods) argument verification from argument parsing so that the argument verification step can return "OK" or "NOT OK" after inspecting the arguments.

Reply:
Agreed.  Will fix.

Reply:
The perfect opportunity to use new booleans in C99 ;-)
I remember I thought of the correct way of handling validation, but the path of least resistance prevailed.
Thank you for keeping me honest!

Reply:
Fixed.
These revisions were fun.  I also included inline struct syntactic sugar mentioned in the article we read.